ultrasonic/src/main/kotlin/org/moire/ultrasonic/playback/LegacyPlaylistManager.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * LegacyPlaylist.kt
+ * Copyright (C) 2009-2022 Ultrasonic developers
+ *
+ * Distributed under terms of the GNU GPLv3 license.
+ */
+
+package org.moire.ultrasonic.playback
+
+import androidx.media3.common.MediaItem
+import androidx.media3.session.MediaController
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.moire.ultrasonic.domain.Track
+import org.moire.ultrasonic.service.DownloadFile
+import org.moire.ultrasonic.service.Downloader
+import org.moire.ultrasonic.service.JukeboxMediaPlayer
+import org.moire.ultrasonic.service.RxBus
+import org.moire.ultrasonic.util.LRUCache
+import timber.log.Timber
+
+/**
+ * This class keeps a legacy playlist maintained which
+ * reflects the internal timeline of the Media3.Player
+ */
+class LegacyPlaylistManager : KoinComponent {
+
+    private val _playlist = mutableListOf<DownloadFile>()
+
+    @JvmField
+    var currentPlaying: DownloadFile? = null
+
+    private val mediaItemCache = LRUCache<String, DownloadFile>(1000)
```

Comment:
So this is where the size of the playlist is limited to 1000.
I've just run into a dead end, and as I see from the comments, you've already been there.
I've tried to remove this cache, and add the DownloadFile directly into the Tag property of the MediaItem.
Unfortunately somehow the LocalConfiguration get stripped from the MediaItems, and the Tag is lost...
I'd really wouldn't like to limit the playlist to 1000. Do you have any better ideas, how this could be handled properly?

Reply:
OK that makes sense now.

 I think to solve it real properly would be to remove this class altogether and rework the Recyclerviews to deal with MediaItems. But that would be too large to put it into this PR.

The easiest fix would probably be to increase the capacity. I just put 1000 because I didn't expect anyone to have that large Playlist. Or try to recreate the DownloadFile based on the values from the MediaItem. But writing this converter would be wasted work later when the class is Refactor Ed away

In the meantime

Reply:
Ok that's what I was thinking too.
I'll increase it to 2000 and continue testing.
(I love to use Ultrasonic as my private radio station with all my favorite songs loaded into it, that's ~1500 songs now, but I also think there are few ppl out there who would load such a list... :)